components/datetime/src/date.rs

```diff
@@ -163,6 +165,30 @@ fn week_of_year<T: DateInput>(
     Ok(WeekOfYear(u32::from(week.week)))
 }
 
+/// Returns the week of month according to a calendar with min_week_days = 1.
+///
+/// This is different from what the UTS35 spec describes but the latter is
+/// missing a month of week-of-month field so following the spec would result
+/// in inconsistencies (e.g. in the ISO calendar 2021-01-01 is the last week
+/// of December but 'MMMMW' would have it formatted as 'week 5 of January').
+fn week_of_month<T: DateInput>(
+    datetime: &T,
+    first_weekday: IsoWeekday,
+) -> Result<WeekOfMonth, DateTimeError> {
+    let day_of_month = datetime
+        .day_of_month()
+        .ok_or(DateTimeError::MissingInput("DateTimeInput::day_of_month"))?;
```

Comment:
Question: When would this error come about?

Reply:
The second case: the user's DateTimeInput implementation returned None for day_of_month(). We now provide a icu_calendar based implementation of DateTimeInput that always returns Some(_) so it doesn't look like case 3 applies here.